src/metabase/driver/hive_like.clj

```diff
@@ -0,0 +1,173 @@
+  (hsql/call :date_format expr (hx/literal format-str)))
+
+(defn- str-to-date [format-str expr]
+  (hx/->timestamp
+   (hsql/call :from_unixtime
+              (hsql/call :unix_timestamp
+                         expr (hx/literal format-str)))))
+
+(defn- trunc-with-format [format-str expr]
+  (str-to-date format-str (date-format format-str expr)))
+
+(defn date [unit expr]
+  (case unit
+    :default expr
+    :minute (trunc-with-format "yyyy-MM-dd HH:mm" (hx/->timestamp expr))
+    :minute-of-hour (hsql/call :minute (hx/->timestamp expr))
+    :hour (trunc-with-format "yyyy-MM-dd HH" (hx/->timestamp expr))
+    :hour-of-day (hsql/call :hour (hx/->timestamp expr))
+    :day (trunc-with-format "yyyy-MM-dd" (hx/->timestamp expr))
+    :day-of-week (hx/->integer (date-format "u"
+                                            (hx/+ (hx/->timestamp expr)
+                                                  (hsql/raw "interval '1' day"))))
+    :day-of-month (hsql/call :dayofmonth (hx/->timestamp expr))
+    :day-of-year (hx/->integer (date-format "D" (hx/->timestamp expr)))
+    :week (hsql/call :date_sub
+                     (hx/+ (hx/->timestamp expr)
+                           (hsql/raw "interval '1' day"))
+                     (date-format "u"
+                                  (hx/+ (hx/->timestamp expr)
+                                        (hsql/raw "interval '1' day"))))
+    :week-of-year (hsql/call :weekofyear (hx/->timestamp expr))
+    :month (hsql/call :trunc (hx/->timestamp expr) (hx/literal :MM))
+    :month-of-year (hsql/call :month (hx/->timestamp expr))
+    :quarter (hsql/call :add_months
+                        (hsql/call :trunc (hx/->timestamp expr) (hx/literal :year))
+                        (hx/* (hx/- (hsql/call :quarter (hx/->timestamp expr))
+                                    1)
+                              3))
+    :quarter-of-year (hsql/call :quarter (hx/->timestamp expr))
+    :year (hsql/call :year (hx/->timestamp expr))))
+
+(defn date-interval [unit amount]
+  (hsql/raw (format "(NOW() + INTERVAL '%d' %s)" (int amount) (name unit))))
+
+(defn string-length-fn [field-key]
+  (hsql/call :length field-key))
+
+;; ignore the schema when producing the identifier
+(defn qualified-name-components
+  "Return the pieces that represent a path to FIELD, of the form `[table-name parent-fields-name* field-name]`."
+  [{field-name :name, table-id :table_id, parent-id :parent_id}]
+  (conj (vec (if-let [parent (metabase.models.field/Field parent-id)]
+               (qualified-name-components parent)
+               (let [{table-name :name, schema :schema} (db/select-one ['Table :name :schema], :id table-id)]
+                 [table-name])))
+        field-name))
+
+(defn field->identifier [field]
+  (apply hsql/qualify (qualified-name-components field)))
+
+;; copied from the Presto driver, except using mysql quoting style
+(defn apply-page [honeysql-query {{:keys [items page]} :page}]
```

Comment:
Same thing for this function and the other ones in this namespace, they either should be private because they're not being used elsewhere, or the need to have a docstring explaining what they're for and how/why someone else might want to use them